cachi2/core/package_managers/pip.py

```diff
@@ -1865,14 +2004,21 @@ def _download_from_requirement_files(
         reference
     :raises PackageRejected: If requirement file does not exist
     """
+    checksums = _get_user_checksums(files, allow_binary)
```

Comment:
This handles checksums on a single-package level. But keep in mind that the user can tell cachi2 to process more than one pip package.

I'm deciding how much we should care about handling it repo-wide vs. keeping it scoped to a single package :thinking:

Reply:
Given that the cachi2 output ends up being a single /deps/pip directory for the entire repo, I'm leaning towards handling it repo-wide

WDYT?